File: fix_settings.py
import re

with open('lib/features/property/presentation/property_detail_screen.dart', 'r') as f:
    content = f.read()

# 1. Update _SettingsTabState build
build_regex = r"""  @override
  Widget build\(BuildContext context\) \{
    final loc = AppLocalizations.of\(context\)[\!];
    final isTr = loc.localeName == 'tr';

    return Column\(
      children: \[
        // ── Toggle ─────────────────────────────────────────────
        Padding\(
          padding: const EdgeInsets.fromLTRB\(24, 20, 24, 0\),
          child: SegmentedButton<String>\(
            segments: \[
              ButtonSegment\(
                value: 'contract',
                label: Text\(isTr \? 'Kontrat Ayarları' : 'Contract Settings', style: const TextStyle\(fontSize: 12\)\),
                icon: const Icon\(LucideIcons.fileEdit, size: 15\),
              \),
              ButtonSegment\(
                value: 'property',
                label: Text\(isTr \? 'Ev Ayarları' : 'Property Settings', style: const TextStyle\(fontSize: 12\)\),
                icon: const Icon\(LucideIcons.home, size: 15\),
              \),
            \],
            selected: \{widget.settingsSectionNotifier.value\},
            onSelectionChanged: \(val\) => widget.settingsSectionNotifier.value = val.first,
            style: SegmentedButton.styleFrom\(
              selectedBackgroundColor: StanomerColors.brandPrimary,
              selectedForegroundColor: Colors.white,
            \),
          \),
        \),

        // ── Panel ──────────────────────────────────────────────
        Expanded\(
          child: SingleChildScrollView\(
            padding: const EdgeInsets.all\(24\),
            child: widget.settingsSectionNotifier.value == 'contract'
                \? _buildContractPanel\(loc, isTr\)
                : _buildPropertyPanel\(loc, isTr\),
          \),
        \),
      \],
    \);
  \}"""

build_replace = """  @override
  Widget build(BuildContext context) {
    final loc = AppLocalizations.of(context)!;
    final isTr = loc.localeName == 'tr';
    final user = ref.watch(currentUserProvider);
    final isLandlord = widget.property.landlordId == user?.id;

    return Column(
      children: [
        // ── Toggle ─────────────────────────────────────────────
        if (isLandlord)
          Padding(
            padding: const EdgeInsets.fromLTRB(24, 20, 24, 0),
            child: SegmentedButton<String>(
              segments: [
                ButtonSegment(
                  value: 'contract',
                  label: Text(isTr ? 'Kontrat Ayarları' : 'Contract Settings', style: const TextStyle(fontSize: 12)),
                  icon: const Icon(LucideIcons.fileEdit, size: 15),
                ),
                ButtonSegment(
                  value: 'property',
                  label: Text(isTr ? 'Ev Ayarları' : 'Property Settings', style: const TextStyle(fontSize: 12)),
                  icon: const Icon(LucideIcons.home, size: 15),
                ),
              ],
              selected: {widget.settingsSectionNotifier.value},
              onSelectionChanged: (val) => widget.settingsSectionNotifier.value = val.first,
              style: SegmentedButton.styleFrom(
                selectedBackgroundColor: StanomerColors.brandPrimary,
                selectedForegroundColor: Colors.white,
              ),
            ),
          ),

        // ── Panel ──────────────────────────────────────────────
        Expanded(
          child: SingleChildScrollView(
            padding: const EdgeInsets.all(24),
            child: (isLandlord && widget.settingsSectionNotifier.value == 'property')
                ? _buildPropertyPanel(loc, isTr)
                : _buildContractPanel(loc, isTr),
          ),
        ),
      ],
    );
  }"""

content = re.sub(build_regex, build_replace, content, 1)

# 2. Update the Tab Title depending on role
tabbar_regex = r"""            Tab\(text: loc.overview\),
            Tab\(text: loc.financials\),
            Tab\(text: loc.activity\),
            Tab\(text: loc.propertySettings\),"""

tabbar_replace = """            Tab(text: loc.overview),
            Tab(text: loc.financials),
            Tab(text: loc.activity),
            Tab(text: currentIsLandlord ? loc.propertySettings : (loc.localeName == 'tr' ? 'Kontrat Ayarları' : 'Contract Settings')),"""

content = re.sub(tabbar_regex, tabbar_replace, content, 1)

with open('lib/features/property/presentation/property_detail_screen.dart', 'w') as f:
    f.write(content)

File: fix_settings.dart
import 'dart:io';

void main() {
  final file = File('lib/features/property/presentation/property_settings_screen.dart');
  String content = file.readAsStringSync();

  content = content.replaceAll('_SettingsTab', 'PropertySettingsScreen');

  content = '''
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:lucide_icons/lucide_icons.dart';
import 'package:go_router/go_router.dart';
import 'package:intl/intl.dart';

import '../../../core/l10n/app_localizations.dart';
import '../../../core/theme/colors.dart';
import '../../property/domain/property.dart';
import '../../property/domain/contract.dart';
import '../../property/domain/expense_item.dart';
import '../../auth/data/auth_providers.dart';
import '../data/property_repository.dart';
import '../data/property_providers.dart';
import '../presentation/widgets/payment_responsibility_selector.dart';

''' + content;

  // Fix constructor
  content = content.replaceFirst(
'''class PropertySettingsScreen extends ConsumerStatefulWidget {
  final Property property;
  final TabController tabController;
  final ValueNotifier<String> settingsSectionNotifier;
  const PropertySettingsScreen({super.key, required this.property, required this.tabController, required this.settingsSectionNotifier});''',
'''class PropertySettingsScreen extends ConsumerStatefulWidget {
  final Property property;
  final String initialTab;
  const PropertySettingsScreen({super.key, required this.property, this.initialTab = 'contract'});'''
  );

  // Fix state class init
  content = content.replaceFirst(
'''class _PropertySettingsScreenState extends ConsumerState<PropertySettingsScreen> {
  void _onSectionChanged() {
    if (mounted) setState(() {});
  }''',
'''class _PropertySettingsScreenState extends ConsumerState<PropertySettingsScreen> {
  late final ValueNotifier<String> settingsSectionNotifier;

  void _onSectionChanged() {
    if (mounted) setState(() {});
  }'''
  );

  content = content.replaceFirst(
'''  @override
  void initState() {
    super.initState();
    widget.settingsSectionNotifier.addListener(_onSectionChanged);''',
'''  @override
  void initState() {
    super.initState();
    settingsSectionNotifier = ValueNotifier(widget.initialTab);
    settingsSectionNotifier.addListener(_onSectionChanged);'''
  );

  content = content.replaceFirst(
'''  @override
  void dispose() {
    widget.settingsSectionNotifier.removeListener(_onSectionChanged);''',
'''  @override
  void dispose() {
    settingsSectionNotifier.removeListener(_onSectionChanged);
    settingsSectionNotifier.dispose();'''
  );

  // Fix build return to be a Scaffold
  content = content.replaceFirst(
'''    final user = ref.watch(currentUserProvider);
    final isLandlord = widget.property.landlordId == user?.id;

    return Column(
      children: [
        // ── Toggle ─────────────────────────────────────────────''',
'''    final user = ref.watch(currentUserProvider);
    final isLandlord = widget.property.landlordId == user?.id;

    return Scaffold(
      appBar: AppBar(
        title: Text(loc.propertySettingsLabel),
      ),
      body: Column(
        children: [
          // ── Toggle ─────────────────────────────────────────────'''
  );

  content = content.replaceFirst(
'''        Expanded(
          child: SingleChildScrollView(
            padding: const EdgeInsets.all(24),
            child: (isLandlord && widget.settingsSectionNotifier.value == 'property')
                ? _buildPropertyPanel(loc)
                : _buildContractPanel(loc),
          ),
        ),
      ],
    );''',
'''        Expanded(
          child: SingleChildScrollView(
            padding: const EdgeInsets.all(24),
            child: (isLandlord && settingsSectionNotifier.value == 'property')
                ? _buildPropertyPanel(loc)
                : _buildContractPanel(loc),
          ),
        ),
      ],
      ),
    );'''
  );

  content = content.replaceAll('widget.settingsSectionNotifier.value', 'settingsSectionNotifier.value');

  file.writeAsStringSync(content);
}
